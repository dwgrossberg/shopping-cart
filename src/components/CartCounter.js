import { useState } from "react";

const CartCounter = (props) => {
  const { price, quantity } = props;
  const handleClick = (e) => {
    console.log(e.target);
  };

  return (
    <div className="price-quant">
      <div className="quant">
        <button className="decrement" onClick={handleClick}>
          -
        </button>
        <div className="quant-number">{quantity}</div>
        <button className="increment" onClick={handleClick}>
          +
        </button>
      </div>
      <div className="price">${price}</div>
    </div>
  );
};

export default CartCounter;
