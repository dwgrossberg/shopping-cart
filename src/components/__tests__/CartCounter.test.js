import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import { screen, render, cleanup } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";
import CartCounter from "../CartCounter/CartCounter";

describe("Cart Counter component", () => {
  afterEach(cleanup);
  const renderer = new ShallowRenderer();
  const user = userEvent.setup();

  it("snapshot test", () => {
    renderer.render(<CartCounter />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("correctly calls item increment counter", async () => {
    const onClick = jest.fn();
    render(
      <div className="price-quant">
        <div className="quant">
          <button className="decrement" onClick={onClick}>
            -
          </button>
          <div className="quant-number">{1}</div>
          <button className="increment" onClick={onClick}>
            +
          </button>
        </div>
        <div className="price">${100}</div>
      </div>
    );
    await user.click(screen.getByText(/-/i));
    expect(onClick).toHaveBeenCalledTimes(1);
    await user.click(screen.getByText(/\+/i));
    expect(onClick).toHaveBeenCalledTimes(2);
  });
});
