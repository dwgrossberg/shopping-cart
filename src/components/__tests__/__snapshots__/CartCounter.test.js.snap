// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Counter component snapshot test 1`] = `
<div
  className="price-quant"
>
  <div
    className="quant"
  >
    <button
      className="decrement"
      onClick={[Function]}
    >
      -
    </button>
    <div
      className="quant-number"
    />
    <button
      className="increment"
      onClick={[Function]}
    >
      +
    </button>
  </div>
  <div
    className="price"
  >
    $
  </div>
</div>
`;
